### DISCUSSION TITLE
Why are checklist text files limited to 37 characters before they line wrap?

### DISCUSSION THREAD

**Discussion by steveveach** [EdgeTX/edgetx#2683] [General]
I was wondering if there was a reason why the checklist text file in the Models folder is limited to 37 characters before line wrapping when the screen real-estate on my RadioMaster TX16S Mark II has more space--probably another 15 characters.

Thanks.

**Comment by pfeerick** [comment]
I think it was done because the maximum number of characters that can fit on a line can't reasonably be determined dynamically, so it is based on how many times the worse-case scenario character - probably w - would fit on the screen. That sound right @eshifri? Think you had fun juggling that one ;)

**Comment by eshifri** [comment]
This is correct. ETX uses TT fonts, so the actual number of characters depend on the characters being used. The code required to estimate the length of the string before loading the block of the text, so the worst case estimation was used. 
Problem is removed in 2.8 by switching to lvgl library (and using far more memory for loading text files :-) ).

**Reply by steveveach** [reply]
Wonderful. Thanks for the explanation. I can look forward to 2.8 then when it's stable for longer lines!

**Reply by raphaelcoeffic** [reply]
LVGL handles the line wrapping by actually looking at what gets actually printed. It should feel more natural.

**Reply by rdeanchurch** [reply]
2.8.4 was a giant step forward for users!  There is still a problem for checklist users because we cannot make a checklist and know how it will be presented on the tx.  A possible solution would be a fixed width character font.  this does not ease reading as kerned fonts look better, but with a fixed width font we would know how many characters the screen would handle.
